drivers/gpio/gpio_hpmicro.c

```diff
@@ -0,0 +1,284 @@
+	 * GPIO module and the corresponding PORT module.
+	 *
+	 * Start with the GPIO module and set up the pin direction register.
+	 * 0 - pin is input, 1 - pin is output
+	 */
+
+	switch (flags & GPIO_DIR_MASK) {
+	case GPIO_DISCONNECTED:
+	case GPIO_INPUT:
+		gpio_disable_pin_output(gpio_base, port_base, pin);
+		break;
+	case GPIO_OUTPUT:
+		gpio_enable_pin_output(gpio_base, port_base, pin);
+		if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			gpio_write_pin(gpio_base, port_base, pin, 1);
+		} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+			gpio_write_pin(gpio_base, port_base, pin, 0);
+		}
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+
+static int gpio_hpm_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+
+	*value = gpio_read_port(gpio_base, port_base);
+
+	return 0;
+}
+
+static int gpio_hpm_port_set_masked_raw(const struct device *dev,
+					 uint32_t mask,
+					 uint32_t value)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+	uint32_t port_val;
+
+	port_val = gpio_read_port(gpio_base, port_base);
+	mask &= gpio_base->OE[port_base].CLEAR;
+	if (mask != 0) {
+		gpio_write_port(gpio_base, port_base, (port_val & ~mask) | (mask & value));
+	}
+
+	return 0;
+}
+
+static int gpio_hpm_port_set_bits_raw(const struct device *dev,
+					   uint32_t mask)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+
+	gpio_set_port_high_with_mask(gpio_base, port_base, mask);
+
+	return 0;
+}
+
+static int gpio_hpm_port_clear_bits_raw(const struct device *dev,
+					 uint32_t mask)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+
+	gpio_set_port_low_with_mask(gpio_base, port_base, mask);
+
+	return 0;
+}
+
+static int gpio_hpm_port_toggle_bits(const struct device *dev, uint32_t mask)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+
+	gpio_toggle_port_with_mask(gpio_base, port_base, mask);
+
+	return 0;
+}
+
+static int gpio_hpm_pin_interrupt_configure(const struct device *dev,
+						 gpio_pin_t pin, enum gpio_int_mode mode,
+						 enum gpio_int_trig trig)
+{
+	const struct gpio_hpm_config *config = dev->config;
+	GPIO_Type *gpio_base = config->gpio_base;
+	uint32_t port_base = config->port_base;
+	gpio_interrupt_trigger_t trigger = 0;
+
+	if (mode == GPIO_INT_MODE_LEVEL) {
+		switch (trig) {
+		case GPIO_INT_TRIG_LOW:
+			trigger = gpio_interrupt_trigger_level_low;
+			break;
+		case GPIO_INT_TRIG_HIGH:
+			trigger = gpio_interrupt_trigger_level_high;
+			break;
+		case GPIO_INT_TRIG_BOTH:
```

Comment:
We're handing `GPIO_INT_TRIG_BOTH` the same way we handle `default`. Can we merge them?